File: 프로그래머스/Level 1/추억 점수.py
# https://school.programmers.co.kr/learn/courses/30/lessons/176963?language=python3
def solution(name, yearning, photo):
    answer = []
    nameDict = {}
    
    for pp in photo:
        nameDict = {} # 임시 딕셔너리
        for p in pp:
            if not p in nameDict:
                nameDict[p] = 1
            else:
                nameDict[p] += 1
        
        score = 0
        for key, value in nameDict.items():
            for i in range(len(name)):
                if name[i] == key:
                    score += value * yearning[i]
                    break
        answer.append(score)
    return answer

File: 프로그래머스/KaKao/추억 점수.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/176963
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    unordered_map<string, int> hashMap;
    
    for (int i = 0; i < name.size(); ++i)
    {
        hashMap[name[i]] = yearning[i];
    }
    
    for (auto &p: photo)
    {
        int tot = 0;
        for (auto n: p)
        {
            tot += hashMap[n];
        }
        answer.push_back(tot);
    }
    return answer;
}